### PR TITLE
fix `Undefined index: eof` #314

### PR DESCRIPTION
**Pull request by max-kut**

### CODE REVIEW

**Comment by sanmai** [comment]
This should be easy to test FWIW.

**Comment by eXorus** [comment]
Do you have a email example that works only with feof() ?

**Comment by max-kut** [comment]
@eXorus yes, I use [horde/Imap_Client](https://github.com/horde/Imap_Client), after using the Fetch command, I pass the resource to the Parser
```
        $res = $imapClient->fetch($query->filter->folder, $this->getFetchQuery(), [
            'ids' => new Horde_Imap_Client_Ids($messagesList->uids),
            'nocache' => true
        ]);

        /** @var \Horde_Imap_Client_Data_Fetch $fetchRes */
        foreach ($res as $uid => $fetchRes) {
            $parser = new Parser();
            $parser->setStream($fetchRes->getFullMsg(true));
            // do work with email
        }
```

**Comment by max-kut** [comment]
When can I see this PR in the release? :-)

**Comment by sanmai** [comment]
I'd like to emphasize that this change could be very well tested with as simple as this:
```php
$fh = fopen(__FILE__, 'r');
while (!feof($fh)) {
    fread($fh, 4096);
}
```

**Comment by eXorus** [comment]
Agree with @sanmai, could be great to add a test for that.

About the release, the master is in the middle of a big release that I'm preparing for the next weeks. Almost a full rewrite of the lib for the next major version v7, with a lot of BC.

So I think it will wait the v7

**Comment by max-kut** [comment]
Thanks for the feedback)

**Comment by eXorus** [comment]
@sanmai How could we test this PR?

```php

        $file = __DIR__.'/mails/message.eml';

        $fh = fopen($file, 'r');
        while (!feof($fh)) {
            fread($fh, 4096);
        }

        $Parser = new Parser();
        $Parser->setStream($fh);

```
Doesn't work we have another error:

> PhpMimeMailParser\Exception: setStream() expects parameter stream to be readable stream resource.

File: src/Parser.php [hunk 1]
```diff
@@ -171,7 +171,7 @@ public function setStream($stream): ParserInterface
     {
         // streams have to be cached to file first
         $meta = @stream_get_meta_data($stream);
-        if (!$meta || !$meta['mode'] || !in_array($meta['mode'], self::$readableModes, true) || $meta['eof']) {
+        if (!$meta || !$meta['mode'] || !in_array($meta['mode'], self::$readableModes, true) || feof($stream)) {
```
Comment (sanmai) [inline src/Parser.php:174]:
```suggestion
        if (!$meta || !$meta['mode'] || !in_array($meta['mode'], self::$readableModes, true)) {
```

**Comment by sanmai** [comment]
That means that we don't need this `eof()` check at all.

**Comment by eXorus** [comment]
Thanks @sanmai and @max-kut 
Merged on master but don't use it right now, work still in progress for the next big version 7.0.0 with a lot of BC that I have to document.

**Pull request merged**